tiledb/sm/serialization/array_schema.cc

```diff
@@ -307,7 +307,15 @@ Status dimension_to_capnp(
 ByteVecValue tile_extent_from_capnp(
     const capnp::Dimension::TileExtent::Reader tile_extent_reader,
     Datatype dim_type) {
+  // When the datatype_size function returns 0, we know that the input datatype
+  // is invalid.
   auto coord_size = datatype_size(dim_type);
+  if (coord_size == 0) {
+    throw std::runtime_error(
+        "Error deserializing tile extent, datatype " + datatype_str(dim_type) +
```

Comment:
What's going to happen when you call `datatype_str` with an invalid argument? There's no string corresponding to that argument. At best you'll get a constant "error string" return.

The string message also needs to say which kind of deserialization is present.

The word "error" is usually redundant in exception messages. It's an exception. A simple declarative sentence is adequate.

Reply:
Fixed in next change.